### PR TITLE
Fixing error in merge where old code in CardBase was readded 

### PR DESCRIPTION
**Pull request by JonaLam**
# Description

Removed an old card effect system which got readded in a merge

## Related issue(s)

#36 

## List of changes

Removed unecesary variables and functions from CardBase
Small changes in CardBase Recources to make some of them work, primaraly heal

### CODE REVIEW

File: Cards/CardBase.gd
Comment (Turtyo) [inline Cards/CardBase.gd:48]: looking at my own PR, this got deleted. i think I put the effect in the `EffectDamage` card. Would like confirmation from someone that reviewed my initial PR (@Tysterman74 or @Tomzkk ) but i'm pretty sure you can also delete this function from CardBase

**Review by Turtyo — COMMENTED** [review]
I will need to check that in local too.
When you modified the cards, did you do that via your text editor or did you use Godot's inspector ? Because looking at some of those resources, it seems like they might be missing some data to work (might just be me, as i said, i didn't test in local yet)

File: Cards/CardBase.gd
Comment (JonaLam) [reply Cards/CardBase.gd:48]: I can see if anything breaks if I remove it

Comment (JonaLam) [reply Cards/CardBase.gd:48]: Ye, it seems like it's fine without so I'll push that

Comment (Turtyo) [reply Cards/CardBase.gd]: perfect thanks !

**Comment by Turtyo** [comment]
I tested in local. It works fine. You should just remove the Card_EnemyAttack.tres from the player deck to not have a card with NULL in hand. Also maybe damage all should be removed since it doesn't do anything currently

File: Cards/CardBase.gd
Comment (Tomzkk) [inline Cards/CardBase.gd:43]: Looking at #27, on_card_play was changed to only have a target and no longer have a caster. Was this change missed or was it at some point changed back to have both ?

**Review by Tomzkk — COMMENTED** [review]
Tried to check by comparing what was removed from here with #27 and it looks like the removed stuff should be okay. Left a comment, I'm not entirely sure if that was changed at some later point or if that's a missing change. If it's missing there might be more like that ?

File: Cards/CardBase.gd
Comment (Turtyo) [reply Cards/CardBase.gd:43]: Good catch, there is no caster in this function for now. Though that will probably change with the introduction of stats

Comment (JonaLam) [reply Cards/CardBase.gd:43]: I did miss it, but if we're gonna use it later it should be fine to just keep it, right?

Comment (Turtyo) [reply Cards/CardBase.gd:43]: Mmh maybe we should check where the function is used to make use don't introduce a bug by keeping it.

**Comment by Turtyo** [comment]
Let's only merge this one after #22 is merged to prevent another merge conflict on an old PR

File: Cards/CardBase.gd
Comment (JonaLam) [reply Cards/CardBase.gd:43]: I think it's only called in battler, and it's not like anything happens with the caster so it shouldn't make any bugs until we use the caster

**Review by Turtyo — APPROVED** [review]

**Review by Tysterman74 — APPROVED** [review]

**Comment by Turtyo** [comment]
#22 was closed to be reopened in a new and cleaner PR, waiting for this new PR to merge #37

**Comment by Turtyo** [comment]
After looking, merging this one now shouldn't give problems with 22 so we'll squash merge now

**Pull request merged**
